apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+#!usr/bin/env tclsh
+
+########################################################################################
+# Name              apixu-weatherbot1.0.tcl
+#
+# Author            Bruce Olivier <bolivierjr@gmail.com>
+#
+# Description       Uses the Apixu weather API to grab the data.
+#
+#                   Requires channel to be flagged with +weather-apixu
+#                       example: .chanset #chan +weather-apixu
+#
+#                   Commands:
+#                       .set 0-2 <location>
+#                          - can be 0 for Metric, 1 for Imperial, 2 for both.
+#                            i.e. ".set 2 New Orleans, LA"
+#                       .wz <location>
+#                          - Grabs current weather location. If no location is
+#                            given, will try to use users saved location
+#
+#                       .wzf <location>
+#                          - Same as .wz but provides a 5 day forecast
+#                       
+#                       Use the --help or -h flags with any of the commands to
+#                       get help with the commands and descriptions.
+#
+# Version           1.0 - Initial Release
+#
+# Notes             Grab your own api key at https://apixu.com and place it in
+#                   the "apikey" variable below. Set private variable to 0 or 1
+#                   if you want the set command responses in PM.
+#
+########################################################################################
+
+namespace eval weather {
+    package require json
+    package require tdom
+    package require http
+    package require tls
+
+    # To be able to query APIs that use https
+    http::register https 443 [list tls::socket -tls1 1 -ssl2 0 -ssl3 0]
+
+    setudef flag weather-apixu
+    # Default units for weather output. 0 for metric, 1 for imperial, 2 for both.
+    variable units "1"
+    variable base_url "https://api.apixu.com/v1"
+    # Set your apikey here
+    variable apikey "<insert-your-key-here>"
+    # Set to 1 (0 by default) if you want set weather responses
+    # to use PM/notify. Set to 0 if you want it all in the channel.
+    variable private 0
+
+    # Public channel command bindings
+    bind pub - .wz weather::current
+    bind pub - .wzf weather::forecast
+    bind pub - .set weather::location
+
+    # Private message command bindings
+    bind msg - .wz weather::pm_current
+    bind msg - .wzf weather::pm_forecast
+    bind msg - .set weather::pm_location
+
+    ##########################
+    # Public command functions
+    ##########################
+    proc current {nick uhost hand chan text} {
+        set text [string trim $text]
```

Comment:
you use this block a lot and might want to change it at one point. maybe you can wrap it into a function call

Reply:
Good idea. I was kind of thinking about that.